### PR TITLE
Enh #8329: Extract messages for specified language

### LINKED ISSUE #8329: Enh: extract messages for specified language
Because translator work mostly with one language, it should be better have ability extract messages for specified language easier than generate new config.php and then edit it.

After this patch translator can use default config.php "from the box" like below:
```
./yii message/extract messages/config.php langCode
```

Patch code:
```php
--- a/framework/console/controllers/MessageController.php
+++ b/framework/console/controllers/MessageController.php
@@ -76,9 +76,12 @@ class MessageController extends Controller
      * @param string $configFile the path or alias of the configuration file.
      * You may use the "yii message/config" command to generate
      * this file and then customize it for your needs.
+     * @param string $specifiedLanguage the language code.
+     * This parameter override 'languages' array content
+     * from $configFile with one specified language code.
      * @throws Exception on failure.
      */
-    public function actionExtract($configFile)
+    public function actionExtract($configFile, $specifiedLanguage = NULL)
     {
         $configFile = Yii::getAlias($configFile);
         if (!is_file($configFile)) {
@@ -93,6 +96,9 @@ class MessageController extends Controller
             'format' => 'php',
             'ignoreCategories' => [],
         ], require($configFile));
+        if (!empty($specifiedLanguage)) {
+            $config['languages'] = [ $specifiedLanguage ];
+        }
 
         if (!isset($config['sourcePath'], $config['languages'])) {
             throw new Exception('The configuration file must specify "sourcePath" and "languages".');
```

### PR DESCRIPTION
**Pull request by vchenin**
close #8329

### CODE REVIEW

File: framework/console/controllers/MessageController.php [hunk 1]
```diff
@@ -76,9 +76,12 @@ public function actionConfig($filePath)
      * @param string $configFile the path or alias of the configuration file.
      * You may use the "yii message/config" command to generate
      * this file and then customize it for your needs.
+     * @param string $specifiedLanguage the language code.
+     * This parameter override 'languages' array content
+     * from $configFile with one specified language code.
      * @throws Exception on failure.
      */
-    public function actionExtract($configFile)
+    public function actionExtract($configFile, $specifiedLanguage = NULL)
```
Comment (samdark) [inline framework/console/controllers/MessageController.php]: `specifiedLanguage` could be renamed to `language`.

Comment (samdark) [inline framework/console/controllers/MessageController.php]: We're using `null`, not `NULL`.

File: framework/console/controllers/MessageController.php [hunk 2]
```diff
@@ -93,6 +96,9 @@ public function actionExtract($configFile)
             'format' => 'php',
             'ignoreCategories' => [],
         ], require($configFile));
+        if (!empty($specifiedLanguage)) {
+            $config['languages'] = [ $specifiedLanguage ];
```
Comment (samdark) [inline framework/console/controllers/MessageController.php]: No need for space after `[` and before `]`.

Comment (samdark) [inline framework/console/controllers/MessageController.php]: Can I specify multiple languages?

Comment (vchenin) [inline framework/console/controllers/MessageController.php]: We can use comma-separated list of language's codes and then explode it in array:
```php
$config['languages'] = explode(",", $languages);
```

File: framework/console/controllers/MessageController.php [hunk 3]
```diff
@@ -76,9 +76,12 @@ public function actionConfig($filePath)
      * @param string $configFile the path or alias of the configuration file.
      * You may use the "yii message/config" command to generate
      * this file and then customize it for your needs.
+     * @param string $languages comma-separated list of language's codes.
+     * This parameter override 'languages' array content
+     * from $configFile with specified language's codes.
      * @throws Exception on failure.
      */
-    public function actionExtract($configFile)
+    public function actionExtract($configFile, $languages = null)
```
Comment (cebe) [inline framework/console/controllers/MessageController.php]: you can use an `array` typehint here to make the explode be done automatically by yii.
Here is an example of how this works: https://github.com/yiisoft/yii2-apidoc/blob/master/commands/ApiController.php#L42

Comment (vchenin) [inline framework/console/controllers/MessageController.php]: @cebe all done

**Pull request closed without merging**